fill/umbilical/proto_build.cc

```diff
@@ -0,0 +1,115 @@
+                break; 
+            case 0b100:
+                rocketMetadata->set_flight_mode(command::MAIN_DEPLOYED);
+                break; 
+            case 0b101:
+                rocketMetadata->set_flight_mode(command::FAULT);
+                break; 
+        }
+
+        events->set_altitude_armed(static_cast<bool>(events_val & 0x1));
+        events->set_altimeter_init_failed(static_cast<bool>((events_val & 0x2) >> 1)); 
+        events->set_altimeter_reading_failed(static_cast<bool>((events_val & 0x4) >> 2)); 
+        events->set_gps_init_failed(static_cast<bool>((events_val & 0x8) >> 3)); 
+        events->set_gps_reading_failed(static_cast<bool>((events_val & 0x10) >> 4)); 
+        events->set_imu_init_failed(static_cast<bool>((events_val & 0x20) >> 5)); 
+        events->set_imu_reading_failed(static_cast<bool>((events_val & 0x40) >> 6)); 
+        events->set_accelerometer_init_failed(static_cast<bool>((events_val & 0x80) >> 7)); 
+        events->set_accelerometer_reading_failed(static_cast<bool>((events_val & 0x100) >> 8)); 
+        events->set_thermometer_init_failed(static_cast<bool>((events_val & 0x200) >> 9)); 
+        events->set_thermometer_reading_failed(static_cast<bool>((events_val & 0x400) >> 10)); 
+        events->set_voltage_init_failed(static_cast<bool>((events_val & 0x800) >> 11)); 
+        events->set_voltage_reading_failed(static_cast<bool>((events_val & 0x1000) >> 12)); 
+        events->set_adc_init_failed(static_cast<bool>((events_val & 0x2000) >> 13)); 
+        events->set_adc_reading_failed(static_cast<bool>((events_val & 0x4000) >> 14)); 
+        events->set_fram_init_failed(static_cast<bool>((events_val & 0x8000) >> 15)); 
+        events->set_fram_write_failed(static_cast<bool>((events_val & 0x10000) >> 16)); 
+        events->set_sd_init_failed(static_cast<bool>((events_val & 0x20000) >> 17)); 
+        events->set_sd_write_failed(static_cast<bool>((events_val & 0x40000) >> 18)); 
+        events->set_mav_was_actuated(static_cast<bool>((events_val & 0x80000) >> 19)); 
+        events->set_sv_was_actuated(static_cast<bool>((events_val & 0x100000) >> 20)); 
+        events->set_main_deploy_wait_end(static_cast<bool>((events_val & 0x200000) >> 21)); 
+        events->set_main_log_shutoff(static_cast<bool>((events_val & 0x400000) >> 22)); 
+        events->set_cycle_overflow(static_cast<bool>((events_val & 0x800000) >> 23)); 
+        events->set_invalid_command(static_cast<bool>((events_val & 0x1000000) >> 24)); 
+
+        rocketUmbTelemetry->set_ms_since_boot(ms_since_boot);
+        rocketUmbTelemetry->set_radio_state(radio_state);
+        rocketUmbTelemetry->set_transmit_state(transmit_state);
+        rocketUmbTelemetry->set_voltage(voltage);
+        rocketUmbTelemetry->set_pt3(pt3);
+        rocketUmbTelemetry->set_pt4(pt4);
+        rocketUmbTelemetry->set_rtd_temp(temp);
+
+    } else {
+        printf("Serial port is not open.");
```

Comment:
In future PRs, you should probably come up with a consolidated error handling scheme (and pass it to the GUI?) instead of just printing it out - this is fine for MVP though

Reply:
Agreed. Created a ticket to track this: https://github.com/cornellrocketryteam/Ground-Software/issues/156